test/common/network_policy_access_ns_to_svc.go

```diff
@@ -0,0 +1,264 @@
+			Name:      serviceName,
+			Namespace: projectName,
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				corev1.ServicePort{
+					Protocol:   corev1.ProtocolTCP,
+					Port:       80,
+					TargetPort: intstr.FromInt(8080),
+				},
+			},
+			Selector: getTestingCurlLabels(),
+		},
+	}
+
+	// check if service exist
+	key := k8sclient.ObjectKey{Name: serviceCR.GetName(), Namespace: serviceCR.GetNamespace()}
+	err := ctx.Client.Get(goctx.TODO(), key, serviceCR)
+	if err != nil && !k8serr.IsNotFound(err) {
+		return nil, fmt.Errorf("error occured while retrieving service %s : %v", serviceCR.GetName(), err)
+	} else if k8serr.IsNotFound(err) {
+		err = openshiftClient.DoOpenshiftCreateServiceInANamespace(openshiftAPIURL, projectName, serviceCR)
+		if err != nil {
+			return nil, fmt.Errorf("error occured while making request to create service %s : %v", serviceCR.GetName(), err)
+		}
+	}
+
+	return serviceCR, nil
+}
+
+func createPodWithAnEndpoint(ctx *TestingContext, openshiftAPIURL string, openshiftClient *resources.OpenshiftClient) (*corev1.Pod, error) {
+
+	podCR := &corev1.Pod{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      podName,
+			Labels:    getTestingCurlLabels(),
+			Namespace: projectName,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				corev1.Container{
+					Name:  containerName,
+					Image: "busybox:1.31.1",
+					Command: []string{
+						"/bin/sh",
+					},
+					Args: []string{
+						"-c",
+						fmt.Sprintf("while true ; do  echo -e \"HTTP/1.1 200 OK\n\n %s \" | nc -l -p 8080  ; done", podEndpoitResponse),
+					},
+					Ports: []corev1.ContainerPort{
+						corev1.ContainerPort{
+							ContainerPort: 8080,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// check if pod exist
+	key := k8sclient.ObjectKey{Name: podCR.GetName(), Namespace: podCR.GetNamespace()}
+	err := ctx.Client.Get(goctx.TODO(), key, podCR)
+	if err != nil && !k8serr.IsNotFound(err) {
+		return nil, fmt.Errorf("error occured while retrieving pod %s : %v", podCR.GetName(), err)
+	} else if k8serr.IsNotFound(err) {
+		err = openshiftClient.DoOpenshiftCreatePodInANamespace(openshiftAPIURL, projectName, podCR)
+		if err != nil {
+			return nil, fmt.Errorf("error occured while making request to create pod %s : %v", podCR.GetName(), err)
+		}
+	}
+
+	return podCR, nil
+
+}
+
+func checkPodStatus(ctx *TestingContext, podCR *corev1.Pod) (bool, error) {
+	checkPodStatus := false
+	key := k8sclient.ObjectKey{Name: podCR.GetName(), Namespace: podCR.GetNamespace()}
+	for !checkPodStatus {
+		err := ctx.Client.Get(goctx.TODO(), key, podCR)
+		if err != nil {
+			return false, fmt.Errorf("error getting pod: %v", err)
+		}
+		for _, cnd := range podCR.Status.Conditions {
+			if cnd.Type == corev1.ContainersReady {
+				if cnd.Status != corev1.ConditionStatus("True") {
+					checkPodStatus = false
+				} else {
+					checkPodStatus = true
+					break
+				}
+			}
+		}
+	}
+	return checkPodStatus, nil
+}
```

Comment:
just a suggestion to shorten this, might be worth wrapping this in a poll, to handle other pod status, like a `crashBackLoop`, an example of a [poll](https://github.com/integr8ly/integreatly-operator/pull/510/files#diff-630e5cb2ca42a372b23875a0a962ee94R120)
```suggestion
	key := k8sclient.ObjectKey{Name: podCR.GetName(), Namespace: podCR.GetNamespace()}
	err := ctx.Client.Get(goctx.TODO(), key, podCR)
	if err != nil {
		return false, fmt.Errorf("error getting pod: %v", err)
	}
	for _, cnd := range podCR.Status.Conditions {
		if cnd.Type == corev1.ContainersReady  && cnd.Status == corev1.ConditionStatus("True") {
			return true, nil
		}
	}
	return false, nil
}
```